### PR TITLE
Fix typos and add Typo check GitHub Action

### PR DESCRIPTION
**Pull request by shirayu**
- Fix typos
- Add typo check GitHub Action: https://github.com/crate-ci/typos

### CODE REVIEW

File: .github/workflows/typos.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: Check typos
+
+on:
+  push:
+  pull_request:
+    branches:
+      - '**'
+  workflow_dispatch:
```
Comment (anton-l) [inline .github/workflows/typos.yml:8]:
```suggestion
  workflow_dispatch:
```
Let's just run it manually for a week to test :)

File: _typos.toml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Files for typos
+# Instruction:  https://github.com/marketplace/actions/typos-action#getting-started
+
+[default.extend-identifiers]
+
+[default.extend-words]
+# Don't correct following words
+nin="nin"
+nd="np" # nd may be np (numpy)
```
Comment (anton-l) [inline _typos.toml:9]: Look like this is what was meant in those docstrings:
```suggestion
nd.array="np.ndarray"
```

File: _typos.toml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Files for typos
+# Instruction:  https://github.com/marketplace/actions/typos-action#getting-started
+
+[default.extend-identifiers]
+
+[default.extend-words]
+# Don't correct following words
+nin="nin"
```
Comment (anton-l) [inline _typos.toml:8]: Could you link a line where this appears please?

**Review by anton-l — COMMENTED** [review]
Pretty cool addition IMO, I'm open to try it in manual mode for now, and then implement nightly/weekly typo-fixing PRs if everything goes well :) Thanks @shirayu! 
@patrickvonplaten @patil-suraj @pcuenca wdyt?

File: _typos.toml (hunk 2 above)
Comment (pcuenca) [reply _typos.toml:9]: I don't understand how this works, from the documentation it looks like it's used to add words to the dictionary (to ignore them), or is it to force corrections? Or both? How was the `nd.array` typo detected in the first place, was it Typo that pointed it out or did @shirayu saw it themselves and so decided to add it here?

**Review by pcuenca — COMMENTED** [review]
It's surprising the number of typos that were detected!

What does the action do when it runs, does it open a PR like this one, or what would the correction workflow look like? (I'm thinking it could just be a part of the Makefile instead, maybe?)

File: _typos.toml (hunk 3 above)
Comment (shirayu) [reply _typos.toml:8]: Hereis the result of `typos` without the line `nin="nin"`.
https://gist.github.com/shirayu/43649cc2a3b32c875c40027c02b0c946

At first I thought `nin` was an abbreviation for something, but is it a typo for `min`?

File: _typos.toml (hunk 2 above)
Comment (shirayu) [reply _typos.toml:8]: I tried typo detection of `np.ndarra` with this change but it did not work.
My understanding is that typos first tokenize the file and then check using the dictionary. 
Therefore, the words that contain `. ` are not recognized well.

File: _typos.toml (hunk 3 above)
Comment (anton-l) [reply _typos.toml]: Looking at https://github.com/huggingface/diffusers/blame/5dda1735fda047f4242d28f91e6e457b9760d52d/src/diffusers/models/resnet.py#L325 
I think `use_in_shortcut` was meant here @patrickvonplaten?

**Comment by patrickvonplaten** [comment]
Very cool! I'd be happy to add it this one since it should be cheap. But we need indeed to be careful that no "correct" words, definitions are accidently changed

File: _typos.toml (hunk 3 above)
Comment (shirayu) [reply _typos.toml]: I fixed `use_in_shortcut` at b75558b.

```diff
-        new_item = new_item.replace("nin_shortcut", "conv_shortcut")
+        new_item = new_item.replace("in_shortcut", "conv_shortcut")
```

Please check whether this change is OK.

Comment (shirayu) [reply _typos.toml]: Another use of `MIN` is [here](https://gist.githubusercontent.com/shirayu/43649cc2a3b32c875c40027c02b0c946/raw/44e2be398f73f72cb5971113754c2e4b85e7c177/NIN2).

I think this should not be changed.
I added a memo about this at b7845a358f580118bf7da6c1c17ec2fa10b6ca50.

**Comment by patrickvonplaten** [comment]
Good for me if you think it's useful @anton-l :-)

**Review by patrickvonplaten — APPROVED** [review]
Happy to follow @shirayu's and @anton-l's lead here

**Comment by anton-l** [comment]
Ok, let's do a trial run then! :)

**Pull request merged**